### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by subhamX**
## Description
The following PR updates the `README.md`.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,8 +8,163 @@ In order to draw conclusions and insights from this dataset, we created the **Li
...
+   └───src # Contains server side Django Apps
+```
+
+
+## Setting Up Local Development Environment Without Docker
+
+### Prerequisites
+
+The frontend application is using react, for which NodeJS and npm are necessary. NodeJS can be installed from [here](https://nodejs.org/en/).
+The backend application is using Django, for which Python necessary. Python can be installed from [here](https://www.python.org/downloads/).
```
Comment (soccerdroid) [inline README.md]: Add the version numbers we are using.

File: README.md [hunk 2]
```diff
@@ -8,8 +8,163 @@ In order to draw conclusions and insights from this dataset, we created the **Li
...
+docker network create --driver=bridge linkedcommons-net
+```
+4. Now run the recently built `linked_commons/backend` image.
+```bash
+docker run --name backend \
+   -p 8000:8000 --env-file ./env \
+   --network=linkedcommons-net \
+   --rm -d linked_commons/backend
+```
+5. Now to start the database in a isolated container.
```
Comment (soccerdroid) [inline README.md]: an*

**Review by soccerdroid — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -8,8 +8,164 @@ In order to draw conclusions and insights from this dataset, we created the **Li
...
+npm start
+```
+4. To create an optimized build for production, run the following command in the terminal.
+```bash
+npm run build
+```
+
+### Backend and Database
+
+
```
Comment (soccerdroid) [inline README.md:68]: delete these empty lines

**Review by soccerdroid — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -8,8 +8,164 @@ In order to draw conclusions and insights from this dataset, we created the **Li
...
+7. Now the backend should be live at `localhost:8000`.
+8. The server needs a running instance of MongoDB. Start the Mongo DB server and ensure that the `authentication credentials` are exactly same as defined in the `.env` file. If you wish to update the data inside the Database, head over to [this](#add-data-to-mongodb) section.
+9. Happy Contributing to Linked Commons! 🚀🚀🚀
+
+
+## Setting Up Local Development Environment using Docker
+
+### Development 
+
+1. Make sure that the root directory contains `docker-compose.yml`. And ensure that `backend/.env` has file is updated and `MONGO_HOSTNAME` is set to `mongodb:27017`.
```
Comment (soccerdroid) [inline README.md]: , and ensure that the `backend/.env` file is updated....

**Review by soccerdroid — CHANGES_REQUESTED** [review]

**Comment by soccerdroid** [comment]
Please can you describe a bit the frontend and backend parts? Maybe as a comment in the directory structure section.  For example:
Backend: django server which contains the API that feeds data to the visualization
Frontend: code that builds and render the visualization in the browser.

**Comment by subhamX** [comment]
> Please can you describe a bit the frontend and backend parts? Maybe as a comment in the directory structure section. For example:
> Backend: django server which contains the API that feeds data to the visualization
> Frontend: code that builds and render the visualization in the browser.

That's a great idea. I have added a few lines in the `Directory Structure` section. Have a look at it.

File: README.md (hunk 4 above)
Comment (soccerdroid) [reply README.md]: please fix this typo

**Review by soccerdroid — CHANGES_REQUESTED** [review]

**Review by soccerdroid — APPROVED** [review]

**Pull request merged**